src/components/tl/ucp/allreduce/allreduce.h

```diff
@@ -8,9 +8,61 @@
 #include "../tl_ucp.h"
 #include "../tl_ucp_coll.h"
 
+enum {
+    UCC_TL_UCP_ALLREDUCE_ALG_KNOMIAL,
+    UCC_TL_UCP_ALLREDUCE_ALG_SRA_KNOMIAL,
+    UCC_TL_UCP_ALLREDUCE_ALG_LAST
+};
+
+#define CHECK_USERDEFINED_OP(_args, _team) do {                         \
+        if (_args.mask & UCC_COLL_ARGS_FIELD_USERDEFINED_REDUCTIONS) {  \
+            tl_error(UCC_TL_TEAM_LIB(_team),                            \
+                     "userdefined reductions are not supported yet");   \
+            status = UCC_ERR_NOT_SUPPORTED;                             \
+            goto out;                                                   \
+        }                                                               \
+    }while(0)
+
+#define CHECK_INPLACE(_args, _team) do {                                \
```

Comment:
I think CHECK_SAME_MEMTYPE is a better name

Reply:
agree